src/PipeSpec.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Mvc\Middleware;
+
+use Closure;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class PipeSpec
+{
+    /**
+     * @var array
+     */
+    private $spec;
+
+    /**
+     * @param string|RequestHandlerInterface|MiddlewareInterface|Closure ...$spec
+     */
+    public function __construct(...$spec)
+    {
+        $this->spec = $spec;
+    }
+
+    public function getSpec(): array
+    {
+        return $this->spec;
+    }
+
+    /**
+     * Support serialization via var_export
+     */
+    public static function __set_state(array $state): self
```

Comment:
`@var array{spec: list<string|RequestHandlerInterface|MiddlewareInterface|Closure>}  $state`

Reply:
I did not bother with those, I plan to introduce psalm as a separate PR and switch to a coding standard v2

Reply:
Yep. That's one possibility. But does it hurt to add them now? 🙂

Reply:
@Xerkus While I think adding psalm support later is a good call, I also think adding the declarations _now_ is good, as it makes it easier to review and understand what the expectations are for properties and arguments. When you do add psalm support, it's also that much less work to do during that step.